src/app/components/search/search.component.spec.ts

```diff
@@ -0,0 +1,147 @@
+import { HarnessLoader } from '@angular/cdk/testing';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { Component, Pipe, PipeTransform } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import { MatAutocompleteModule } from '@angular/material/autocomplete';
+import { MatAutocompleteHarness } from '@angular/material/autocomplete/testing';
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatIconHarness } from '@angular/material/icon/testing';
+import { MatInputModule } from '@angular/material/input';
+import { MatInputHarness } from '@angular/material/input/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { SearchComponent } from './search.component';
+import { SearchHarness } from './search.harness';
+
+@Pipe({ name: 'searchOptions' })
+class MockPipe implements PipeTransform {
+  transform(value: any): any {
+    return value;
+  }
+}
+
+@Component({
+  selector: 'tk-test',
+  template:
+    '<tk-search [options]="options" (valueChange)="valueChange($event)"></tk-search>'
+})
+class TestComponent {
+  options: string[];
+
+  valueChange(event): void {}
```

Comment:
And again, don't create real methods when you are using _**spyOn**_ instead of them.